ic-os/components/ic/generate-replica-config.sh

```diff
@@ -155,6 +157,15 @@ sed -e "s@{{ ipv6_address }}@${IPV6_ADDRESS}@" \
     -e "s@{{ jaeger_addr }}@${JAEGER_ADDR}@" \
     "${IN_FILE}" >"${OUT_FILE}"
 
+# Generate and inject a self-signed TLS certificate and key for ic-boundary
+# for the given domain name. To be used in system tests only.
+if [[ -n "${GENERATE_IC_BOUNDARY_TLS_CERT}" ]] && [ "${GENERATE_IC_BOUNDARY_TLS_CERT}" != "null" ]; then
+    openssl req -x509 -newkey rsa:2048 \
+        -keyout /var/lib/ic/ic-boundary-tls.key \
+        -out /var/lib/ic/ic-boundary-tls.crt -sha256 -days 3650 -nodes \
+        -subj /C=CH/ST=Zurich/L=Zurich/O=InternetComputer/OU=ApiBoundaryNodes/CN=${GENERATE_IC_BOUNDARY_TLS_CERT}
+fi
+
```

Comment:
This could exist as a separate service, or if we're fine leaving it in generate-replica-config.sh, that works for me too